server/src/main/java/org/candlepin/async/LoggingJob.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.async;
+
+import org.candlepin.pinsetter.core.model.JobStatus;
+import org.slf4j.MDC;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Stores the job's unique ID and other information in logger's thread local MDC, which will
+ * automatically add it to all log entries executed for this job.
+ */
+public class LoggingJob implements AsyncJob {
+
+    private final AsyncJob job;
+
+    public LoggingJob(final AsyncJob job) {
+        this.job = requireNonNull(job);
+    }
+
+    @Override
+    public Object execute(final JobExecutionContext jdata) throws JobExecutionException {
```

Comment:
I think this should be setting the CorrelationId from the AsyncJobStatus on the logging context. Perhaps this should be done somewhere else though.

Reply:
Depends on whether we want CorellationId for the whole request or just the job execution. I think it should be set when the message is received.

Reply:
I may be misunderstanding here, but the objective of the correlation ID/correlated service ID is to track all Candlepin operations (log output) associated with a given operation from the client. As part of that goal, the CSID should be set during the execution of a job that was created from a request with a non-null CSID.

Reply:
That's what I mean. Message is part of the user request so CSID should be set when the message is received right?

Reply:
Kind of. When we queue a job, if there's a CSID set in the MDC, then we pull it from there and put it in the JobStatus. From there, when the job is executed, we need to set it again as the job is running in a different context.